### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by Itxaka**
Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.17-alpine AS golang
+FROM gcr.io/projectsigstore/cosign:v1.4.0 AS cosign-bin
```
Comment (mudler) [inline Dockerfile]: I'd use our images for cosign here from the cos releases repo, they just contain the binary and it's tested and built by our CI

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+
+FROM golang as elemental-bin
+WORKDIR /src/
+COPY . /src/
+RUN CGO_ENABLED=0 go build -o /usr/bin/elemental
+
+FROM alpine AS elemental-alpine
```
Comment (mudler) [inline Dockerfile]: we should stick with opensuse here imho (besides obvious reasons), we have less chance to have incompatibilities and failures due to binary which are misaligned in versions.

File: Dockerfile (hunk 1 above)
Comment (Itxaka) [reply Dockerfile]: Problem with that is our version scheme is not that nice with all the -number stuff :/

File: Dockerfile (hunk 2 above)
Comment (Itxaka) [reply Dockerfile]: Yep, this as mentioned is a PPC and wanted to see if it was possible to build a valid image from alpine as I know it's possible from an opensuse base.

After all the size difference is not that much, ends up about 50Mb bigger so..

File: Dockerfile (hunk 1 above)
Comment (mudler) [reply Dockerfile]: Why? it is just something like:

```
quay.io/costoolkit/releases-green:cosign-toolchain-1.4.1
```

**Review by mudler — APPROVED** [review]
Looks good here!

**Pull request merged**